numba/cuda/simulator/cudadrv/devicearray.py

```diff
@@ -30,23 +30,54 @@ class FakeShape(tuple):
     indexing, similar to the shape in CUDA Python. (Numpy shape arrays allow
     negative indexing)
     '''
+
     def __getitem__(self, k):
         if isinstance(k, int) and k < 0:
             raise IndexError('tuple index out of range')
         return super(FakeShape, self).__getitem__(k)
 
 
+class FakeKernelCUDAArray(object):
+    def __init__(self, item):
+        assert isinstance(item, FakeCUDAArray)
+        self.__dict__['_item'] = item
+
+    def __wrap_if_fake(self, item):
+        return FakeKernelCUDAArray(item) if isinstance(item, FakeCUDAArray) else item
+
+    def __getattr__(self, attrname):
+        if attrname in dir(self.__dict__['_item']._ary):  # For, eg, array size.
```

Comment:
You can use `self._item` here, no need to go through `__dict__`.

Reply:
Can you though? Because this is in self's __getattr__ function and if you ask self for '_item', it'll go right back to where it is.

Reply:
getattr is not called for entries present in `__dict__` (unlike setitem which is)